descope/internal/mgmt/sso.go

```diff
@@ -146,20 +174,45 @@ func (s *sso) GetSettings(ctx context.Context, tenantID string) (*descope.SSOSet
 	return unmarshalSSOSettingsResponse(res)
 }
 
-func (s *sso) DeleteSettings(ctx context.Context, tenantID string) error {
+func (s *sso) NewSettings(ctx context.Context, tenantID string, ssoID string, displayName string) (*descope.SSOTenantSettingsResponse, error) {
+	if tenantID == "" {
+		return nil, utils.NewInvalidArgumentError("tenantID")
+	}
+
+	if displayName == "" {
+		return nil, utils.NewInvalidArgumentError("displayName")
+	}
+
+	req := map[string]any{
+		"tenantId":    tenantID,
+		"ssoId":       ssoID,
+		"displayName": displayName,
+	}
+	res, err := s.client.DoPostRequest(ctx, api.Routes.ManagementNewSSOSettings(), req, nil, s.conf.ManagementKey)
+	if err != nil {
+		return nil, err
+	}
+	return unmarshalSSOTenantSettingsResponse(res)
+}
+
+func (s *sso) DeleteSettings(ctx context.Context, tenantID string, ssoID string) error {
 	if tenantID == "" {
 		return utils.NewInvalidArgumentError("tenantID")
 	}
 	req := &api.HTTPRequest{
 		QueryParams: map[string]string{"tenantId": tenantID},
 	}
+	if len(ssoID) > 0 {
```

Comment:
This should be mandatory no ?
If empty, do we allow to delete the default ?

Reply:
This is how it worked until now, it changes default to none